### ISSUE TITLE
Module 1 Feedback

### ISSUE THREAD

**Issue by hfboyce** [UBC-MDS/introduction-machine-learning#18]
Here is the first module for the content side. -> https://intro-machine-learning.netlify.app/
This is a mix of module 0 and 1 of Varada's material with a bit of transcription and clarifications from me. 
I did not add a part about regression in this module but I added it in the assignment. Do you think I should add a similar few slide of DummyRegressor in after DummyClassifier?

**Comment by mgelbart** [comment]
My feedback (cc @kvarada )

- 0.2: delete "be able to"
- 1.2: images are cut off for me at the bottom
- 1.5: per my comment in the meeting, can we do some of the preprocessing in advance? including train/test split perhaps. Or, is it possible to have code blocks that don't get shown as part of the slides? that would be less work and more reproducible.
- 1.6: the output is fairly meaningless here - suppress it with `;` at the end of the last line
- 1.7: for an intuitive introduction, I think it's more intuitive if we don't have `y_test`
- 1.9: which model is this? maybe more than thousands of images?
- 1.10: i've starting called this a "probability score" in CPSC 330, to emphasize it's really not a probability. not sure if we want to follow suit here and/or in 571 @kvarada ? I tell them, "just because this says 90% probability, please don't go to your boss and say YOU are 90% confident!" because these probability scores can be arbitrarily tweaked by changed hyperparameters, for example.
- 1.12: same comment about the wrangling
- 1.13: `X_test, y_test = train_df['review'], train_df['label']` whoops! ;; also same thing about the semicolon and the output.
- 1.14: `prediction": y_test[0:4],` this should not be called prediction - maybe true sentiment?
- All these are supervised learning. I think maybe we should address supervised vs. unsupervised before this? It just seems a bit off to have the title "What is Machine Learning" showing up through all these examples. Maybe "What is Supervised Learning" ?
- 1.16: I would go for more separation of X and y earlier on. Can we separate them out first and then show them separately? it's very important to highlight what we're trying to predict. 
- 1.17: semicolon;; no need for the first import?
- 1.19: this is a lot of text - maybe this is something we could move to the end of the course and elaborate more on once they are more up to speed. 
- 2: fix title of exercise. speaking of which, maybe we should introduce the acronym ML early on and just use that all the time?
- 2.1: no capitalization on the M
- 2.3: 😂 
- 2: again, these are all supervised learning. UPDATE: But OK, I see the structure now. I guess I can live with calling everything "machine learning" for 2 exercises, rather than have you restructure everything. 
- 3.3: this is a bit misleading, as it seems like you need to call `fit` every time you call `predict`. can we move the unseen example and prediction below, and then have two arrows (one from ML model and one from unseen example) pointing into the prediction?
- 3.8: worth saying that sometimes it's to save time, and other times to gets better than human performance;; don't introduce the "f" notation
- 4.4: is the answer wrong?
- 4.5: unclickable
- 5.2: "discrete classes" -> "categories, also known as classes";; delete "typically floating point" and add "in other words, a number";; add a classification example with more than 2 classes
- 6: add a classification example with more than 2 classes
- 7.2: let's cut this down to just examples/rows and features/inputs and targets/outputs and training/learning/fitting. ;; not sure if we'll need the "n" and "d" in this course, I don't use them in CPSC 330. But I'm not too bothered by leaving them in, meh.
- 7.5: the drop syntax seems a bit unusual to me, is the `axis=1` needed? Is that how we showed it in the other course?
- 9: change outcomes to outputs
- 10: after the changes to 7.2, change this to which IS a synonym and include only one right answer (inputs)
- 11: small "D" in dataset;; we should say what we're trying to predict here, otherwise the last question can't be answered;; I think the answer might be incorrect, isn't it 9?  
- 11-12: no need for the numpy import right? 
- 13.2: see earlier comment about this diagram
- 13.3: I'm a bit confused about the uniform baseline - what is this for? UPDATE: it seems this is a real thing in sklearn. seems kind of weird to me, I'm not sure it's worth showing.
- 13.5: I avoid the term "vectors" in CPSC 330... maybe "feature table" and "targets"?
- 13.7: semicolon
- 13.8: I think it's easier to first show them a prediction on one example before showing this bulk prediction 
- 13.8-13.10: we should say explicitly that we're predicting/scoring on the same data we used for training, but that will change very soon
- 13.10: Step 2 should be the constructor, e.g. `clf = DummyClassifier()`;; this isn't quite right hmm how about making step 4 as `clf.score(X,y)` and then step 5 as predicting on some new data, e.g. `clf.predict(Xnew)`. That is a bit more faithful to the real steps.
- 14.2: not answerable with the information given (decision trees not covered yet)
- 15: let's delete this, I don't think it adds value. Instead we could have one where we show the output of `y.value_counts()` and ask them what dummyclassifier would predict.
- 16: title needs fixing. seems like some sort of freudian slip.;; see above, can we stick to `most_frequent` (or `prior`, which is the same as `most_frequent` for `predict` but different for `predict_proba`, I'm fine either way);; numpy import not needed again I think
- 17.2: see comments for 0.2

FYI @kvarada this took me 30 min. However, this felt significantly shorter than the typical module. It usually takes me an hour or more. That said, this material is more mature to begin with. That said, it may take longer because this is your first time doing this. Long story short, I think you might find yourself spending ~1 hr on reviewing the interactive lesson portion for one module.

**Comment by mgelbart** [comment]
An additional comment now that I'm looking at the assignment: we should clearly state that `DummyClassifier` doesn't actually look at `X` because it's sort of a silly special case, but everything else does look at X and we'll be getting there soon.

**Comment by kvarada** [comment]
@hfboyce I have pushed the .key source file for all images here: 
https://github.ubc.ca/MDS-2020-21/DSCI_571_sup-learn-1_instructors/blob/master/lectures/images/571_lectures_images_source.key

**Comment by kvarada** [comment]
> i've starting called this a "probability score" in CPSC 330, to emphasize it's really not a probability. not sure if we want to follow suit here and/or in 571 @kvarada ? I tell them, "just because this says 90% probability, please don't go to your boss and say YOU are 90% confident!" because these probability scores can be arbitrarily tweaked by changed hyperparameters, for example.

I agree. Calling it a probability score or a normalized score is a good idea.

**Comment by kvarada** [comment]
Thanks @mgelbart ! Your feedback is also helping me to fix/improve 571 lectures. 

> 3.3: this is a bit misleading, as it seems like you need to call fit every time you call predict. can we move the unseen example and prediction below, and then have two arrows (one from ML model and one from unseen example) pointing into the prediction?

I agree. That image is misleading. Probably this one is better? 
<img width="1594" alt="sup-learning" src="https://user-images.githubusercontent.com/22737743/94993818-65e19980-0548-11eb-9e65-ec0b64c823fd.png">

@hfboyce I have updated the image in [the lecture notes](https://github.ubc.ca/MDS-2020-21/DSCI_571_sup-learn-1_instructors/blob/master/lectures/images/sup-learning.png).

**Comment by mgelbart** [comment]
Nice. I would actually move the "unseen test examples" over one more space to the right. The unseen examples are not used to produce _f_, but rather we combine _f_ with the Xtest to get the yhat. So we'd have two arrows pointing into yhat rather than two arrows pointing into _f_. Maybe I'd also move the unseen test examples above yhat rather than below, because we tend to think of the lower-left element of a diagram as the final product.

**Comment by kvarada** [comment]
Oh I see what you meant. I guess one problem is that there are no labels on the arrows. I am not adding labels to the arrows but hopefully the following version is clearer. The idea of "applying" the model on the test data is not super clear in this diagram but I guess two arrows pointing to X_test conveys the idea.  

<img width="1689" alt="sup-learning" src="https://user-images.githubusercontent.com/22737743/95000430-9ab81580-0575-11eb-8dec-9ad69f49dba7.png">

**Comment by mgelbart** [comment]
Yeah, this is what I had in mind. I guess you could put in a "prediction algorithm" somewhere but I think this version is fairly clear.

**Comment by hfboyce** [comment]
@kvarada Can you send me the pptx file please?

**Comment by kvarada** [comment]
Oh sorry, @hfboyce . I have pushed updated .key and .pptx files in the repo. 

https://github.ubc.ca/MDS-2020-21/DSCI_571_sup-learn-1_instructors/blob/master/lectures/images/571_lectures_images_source.key

https://github.ubc.ca/MDS-2020-21/DSCI_571_sup-learn-1_instructors/blob/master/lectures/images/571_lectures_images_source.pptx
